modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,881 @@
+            "        {\n" +
+            "          \"$ref\": \"https://localhost:9443/scim2/Users/3a12bae9-4386-44be-befd-caf349297f45\",\n" +
+            "          \"display\": \"kim\",\n" +
+            "          \"value\": \"008bba85-451d-414b-87de-c03b5a1f4217\"\n" +
+            "        }\n" +
+            "      ],\n" +
+            "      \"id\": \"" + GROUP_ID + "\"\n" +
+            "    }\n" +
+            "  ]\n" +
+            "}";
+
+    private static final String NEW_GROUP_SCIM_OBJECT_STRING_UPDATED = "{\n" +
+            "  \"schemas\": [\n" +
+            "    \"urn:ietf:params:scim:api:messages:2.0:ListResponse,\"\n" +
+            "  ],\n" +
+            "  \"displayName\": \"manager_sales\",\n" +
+            "      \"meta\": {\n" +
+            "       \"created\": \"2019-08-26T14:27:36\",\n" +
+            "        \"location\": \"https://localhost:9443/scim2/Groups/7bac6a86-1f21-4937-9fb1-5be4a93ef469\",\n" +
+            "        \"lastModified\": \"2019-08-26T14:28:36\"\n" +
+            "      },\n" +
+            "      \"members\": [\n" +
+            "        {\n" +
+            "          \"$ref\": \"https://localhost:9443/scim2/Users/3a12bae9-4386-44be-befd-caf349297f45\",\n" +
+            "          \"display\": \"kim\",\n" +
+            "          \"value\": \"008bba85-451d-414b-87de-c03b5a1f4217\"\n" +
+            "        }\n" +
+            "      ],\n" +
+            "      \"id\": \"" + GROUP_ID + "\"\n" +
+            "    }\n" +
+            "  ]\n" +
+            "}";
+
+    private GroupResourceManager groupResourceManager;
+    private UserManager userManager;
+
+    @BeforeMethod
+    public void setUp() {
+
+        groupResourceManager = new GroupResourceManager();
+        userManager = mock(UserManager.class);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+    }
+
+    private SCIMResponse getEncodeSCIMExceptionObject(AbstractCharonException exception) {
+
+        JSONEncoder encoder = new JSONEncoder();
+        Map<String, String> responseHeaders = new HashMap<>();
+        responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+        return new SCIMResponse(exception.getStatus(), encoder.encodeSCIMException(exception), responseHeaders);
+    }
+
+    private Group getNewGroup() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+    }
+
+    @DataProvider(name = "dataForGetGroupSuccess")
+    public Object[][] dataToGetGroupSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        Group group = getNewGroup();
+        String id = group.getId();
+
+        return new Object[][]{
+                {id, null, null, group},
```

Comment:
no need to pass `id` here as a separate param as it can be retrieved from `group`.

Reply:
yes agreed, but i believe this way is more readable since original ``get`` method definition is ``public SCIMResponse get(String id, UserManager userManager, String attributes, String excludeAttributes)`` and all its parameters coming from its test definition `` public void testGetGroupSuccess(String id, String attributes, String excludeAttributes, Object objectGroup) throws BadRequestException, NotFoundException, CharonException, NotImplementedException``  And test gets the data from data provider  `` {id, "displayName", null, group}`` , So this gives more visibility to the data flow and i believe its easy to modify the test if required.  

Since passing ``id`` doesn't affect with performance of the test case and doesn't violate any best practice standard. Let;s pass this way since its no harm.